### DISCUSSION TITLE
Migrate history database from DNF 4

### DISCUSSION THREAD

**Discussion by lantw44** [rpm-software-management/dnf5#1846] [Ideas]
The history database is one of important reasons I use Fedora on laptops. It is very valuable since it logs every change since the system was first installed. When I forget why a package is installed or removed, I just type `dnf history list`. Most of the time, I can quickly figure out the reason from the command line or packages altered in the transaction.

However, it looks like we may lose access to old history data. Since DNF 5 has a separate database, we need to be able to migrate the DNF 4 history database to the DNF 5 history database, or we will have no easy tool to query the history database after DNF 4 is removed. I can still query packages installed with Fedora 16 YUM on Fedora 40 thanks to the history database migration feature of early DNF. Here is the proof of the migration in my Bash history:
```
192766  2015-07-04 17:12:36 sudo dnf-2 migrate 
```
Now yum-deprecated no longer exists in Fedora repositories, but I can still read history written by YUM with DNF 4 since it has been migrated to the DNF history database. I hope we will soon be able to read history written by both YUM and old DNF with DNF 5 with some migration feature.

**Comment by jan-kolarik** [comment]
Hi @lantw44, thanks for the feedback. This is indeed an issue that is affecting various users after migrating to dnf5. I've created a [dedicated issue](https://github.com/rpm-software-management/dnf5/issues/1871) to discuss and implement this further.
